### DISCUSSION TITLE
Debugging kopf operator in VSCode

### DISCUSSION THREAD

**Discussion by mirza-ali-ctct** [nolar/kopf#978] [Q&A]
I am trying to debug `kopf` operator in VSCode and looking for any examples. Is there any way to configure kopf debugging in VSCode IDE.

**Comment by saleem8054** [comment]
me 2

**Comment by saleem8054** [comment]
ok so the question is asked very long time ago but anyway, if someone like me stumbles upon this question and comes here, so here is the answer for you.

In your launch.json, paste this
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Python Debugger: Current File with Arguments",
            "type": "debugpy",
            "request": "launch",
            "console": "integratedTerminal",
            "program": "${workspaceFolder}/.venv/bin/kopf",
            "args": [
                "run",
                "/path/to/your/controller.py" 
            ],
            "env": {
                "KOPF_RUN_ARGS": "--verbose"
            },

        }
    ]
}
